blender/arm/material/make_mesh.py

```diff
@@ -536,15 +542,25 @@ def make_forward(con_mesh):
                 frag.add_uniform('sampler2DShadow shadowMapSpot[4]', included=True)
 
     if not blend:
-        mrt = rpdat.rp_ssr  # mrt: multiple render targets
-        if mrt:
+        mrt = 0
+        if rpdat.rp_ssr:
+            mrt = 2
+        if rpdat.rp_ss_refraction:
+            mrt += 1
+        if mrt != 0:
             # Store light gbuffer for post-processing
-            frag.add_out('vec4 fragColor[2]')
+            frag.add_out('vec4 fragColor[{mrt}]'.format(mrt = mrt))
```

Comment:
You can simplify this by using `frag.add_out(f'vec4 fragColor[{mrt}]')` (note the `f` before the string literal)

Reply:
Ok